ibis/backends/__init__.py

```diff
@@ -838,6 +843,31 @@ def _finalize_cached_table(self, name: str) -> None:
                 if not sys.is_finalizing():
                     raise
 
+    def _should_cache_table_expr(self, expr: ir.Table) -> bool:
+        """Checks if a given table expression is worth caching."""
+        op = expr.op()
+
+        # Don't cache if an expression is a column subselection of a physical table.
+        while isinstance(op, (ops.Project, ops.DropColumns)):
+            if isinstance(op, ops.Project) and not all(
+                isinstance(v, ops.Field) for v in op.values.values()
+            ):
```

Comment:
Most of this code is unrelated to the backend and is about some property of an expression.

Can we define this as an attribute on `ops.Relation` subclasses? Seems like it'd be much less squishy that way.

Reply:
Meaning the decision is consistent and only based on the operation type (no query in the backend needed)? That's option 2 listed above.